commands/drains/index.js

```diff
@@ -10,11 +11,22 @@ function* run (context, heroku) {
   if (context.flags.json) {
     cli.log(JSON.stringify(drains, null, 2));
   } else {
-    drains.forEach(function (drain) {
-      let output = `${cli.color.cyan(drain.url)} (${cli.color.green(drain.token)})`;
-      if (drain.extended) output = output + ` drain_id=${drain.extended.drain_id}`;
-      cli.log(output);
-    });
+    drains = _.partition(drains, 'addon');
+    if (drains[1].length > 0) {
+      cli.styledHeader('Drains');
+      for (let drain of drains[1]) {
+        let output = `${cli.color.cyan(drain.url)} (${cli.color.green(drain.token)})`;
+        if (drain.extended) output = output + ` drain_id=${drain.extended.drain_id}`;
+        cli.log(output);
+      }
+    }
+    if (drains[0].length > 0) {
+      cli.styledHeader('Add-on Drains');
+      for (let drain of drains[0]) {
+        let addon = yield heroku.get(`/apps/${context.app}/addons/${drain.addon.name}`);
```

Comment:
I believe doing `addons = yield drains.map(function(drain) { return heroku.get('/apps/...); })` should perform the requests in parallel at least.